cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,680 @@
+
+}
+
+func promptAzureIntegrationNameQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionAzureConfigName, Default: config.ConfigIntegrationName},
+			Checks:   []*bool{&config.Config},
+			Required: true,
+			Response: &config.ConfigIntegrationName,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionActivityLogName, Default: config.ActivityLogIntegrationName},
+			Checks:   []*bool{&config.ActivityLog},
+			Required: true,
+			Response: &config.ActivityLogIntegrationName,
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func promptAzureStorageAccountQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionUseExistingStorageAccount, Default: config.ExistingStorageAccount},
+			Response: &config.ExistingStorageAccount,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionAwsRegion, Default: config.StorageLocation},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageLocation,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionStorageAccountName, Default: config.StorageAccountName},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageAccountName,
+		},
+		{
+			Prompt:   &survey.Input{Message: QuestionStorageAccountResourceGroup, Default: config.StorageAccountResourceGroup},
+			Checks:   []*bool{&config.ExistingStorageAccount},
+			Required: true,
+			Response: &config.StorageAccountResourceGroup,
+		},
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func promptAzureSubscriptionQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionEnableAllSubscriptions, Default: config.AllSubscriptions},
+			Response: &config.AllSubscriptions,
+		},
+	}); err != nil {
+		return err
+	}
+	var idList string
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Input{Message: QuestionSubscriptionIds, Default: strings.Join(config.SubscriptionIds, ",")},
+			Checks:   []*bool{allSubscriptionsDisabled(config)},
+			Required: true,
+			Response: &idList,
+		},
+	}); err != nil {
+		return err
+	}
+	config.SubscriptionIds = strings.Split(strings.ReplaceAll(idList, " ", ""), ",")
+
+	return nil
+}
+
+func promptAzureManagementGroupQuestions(config *azure.GenerateAzureTfConfigurationArgs) error {
+
+	if err := SurveyMultipleQuestionWithValidation([]SurveyQuestionWithValidationArgs{
+		{
+			Prompt:   &survey.Confirm{Message: QuestionEnableManagementGroup, Default: config.ManagementGroup},
```

Comment:
I'm trying to understand this management group option. If this is set, do you need to provide subscription ids also, or is this saying that we're integrating at a level above subscription?
It might be nice to call out that this option only applicable to the config integration

Reply:
My understanding is that if the management group is set then the active directory will use the management group level rather than the subscription level, so subscription information is superfluous

Reply:
Hmm, guess this is a tricky one because it only applies to the config integration 🤔